src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -174,42 +178,42 @@ export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
   // Cancel button in "Add an Amalgamating Business" is pressed.
   addAmalgamatingBusinessCancel (): void {
     this.isAddingAmalgamatingBusiness = false
+    this.setAmalgamatingBusinessesValid(true)
   }
 
   // "Add an Amalgamating Business" button is pressed.
   onAddBusinessClick (): void {
     this.isAddingAmalgamatingBusiness = true
     this.isAddingAmalgamatingForeignBusiness = false
+    this.setAmalgamatingBusinessesValid(false)
   }
 
   // "Add an Amalgamating Foreign Business" button is pressed.
   onAddForeignBusinessClick (): void {
     this.isAddingAmalgamatingBusiness = false
     this.isAddingAmalgamatingForeignBusiness = true
+    this.setAmalgamatingBusinessesValid(false)
   }
 
   async saveAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
-    // Get the amalgamating business information
-    // Will have a different format depending on the business
-    let business = await LegalServices.fetchBusinessInfo(businessLookup.identifier)
-      .then((response) => {
-        return response?.data?.business
-      }).catch(() => {
-        return businessLookup
-      })
-
-    // Get the address of the amalgamating business
-    if (businessLookup.identifier && business.foundingDate) {
-      const addresses = await LegalServices.fetchAddresses(businessLookup.identifier)
-        .then((data) => {
-          // SP and GP have businessOffice instead of registeredOffice
-          return data?.registeredOffice || data?.businessOffice
-        }).catch(() => {
-          return undefined
-        })
-      if (addresses) {
-        business.officeAddress = addresses
-      }
+    let business = null
+
+    // Get the amalgamating business information, mailing address, and email if in LEAR.
+    // Otherwise, return the businesslookup object.
+    const data = await Promise.all([
+      LegalServices.fetchBusinessInfo(businessLookup.identifier),
+      AuthServices.fetchAuthInfo(businessLookup.identifier),
+      LegalServices.fetchAddresses(businessLookup.identifier)
+    ]).catch((error) => {
+      return error
+    })
+
+    if (data.length === 3) {
+      business = data[0].data?.business
+      business.businessContact = data[1].contacts[0]
+      business.officeAddress = data[2]
+    } else {
+      business = businessLookup
```

Comment:
I don't think we want to do anything with the business if we can't look it up...?

Reply:
True, but we still can get the name, identifier, and status. I was thinking that might help in filling in the table (if needed).

Reply:
Check with Yui or Mihai. I _think_ that if we don't find the business in Auth/Legal APIs then we don't add it to the table. But I'm thinking we might want to let the user know somehow...

Reply:
Removed the else block.

The business variable will still be null if COLIN entity.